examples/shared-textures/main.rs

```diff
@@ -0,0 +1,339 @@
+    window: Window,
+    surface: wgpu::Surface,
+    sc_desc: wgpu::SwapChainDescriptor,
+    swap_chain: wgpu::SwapChain,
+}
+
+impl Viewport {
+    fn new(
+        window: Window,
+        device: &wgpu::Device,
+        swapchain_format: wgpu::TextureFormat,
+        instance: &wgpu::Instance,
+    ) -> Self {
+        let surface = unsafe { instance.create_surface(&window) };
+        Self::with_surface(window, device, swapchain_format, surface)
+    }
+    fn with_surface(
+        window: Window,
+        device: &wgpu::Device,
+        swapchain_format: wgpu::TextureFormat,
+        surface: wgpu::Surface,
+    ) -> Self {
+        let size = window.inner_size();
+
+        let sc_desc = wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: swapchain_format,
+            width: size.width,
+            height: size.height,
+            present_mode: wgpu::PresentMode::Fifo,
+        };
+
+        let swap_chain = device.create_swap_chain(&surface, &sc_desc);
+
+        Viewport {
+            window,
+            surface,
+            sc_desc,
+            swap_chain,
+        }
+    }
+    fn resize(&mut self, device: &wgpu::Device, size: winit::dpi::PhysicalSize<u32>) {
+        self.sc_desc.width = size.width;
+        self.sc_desc.height = size.height;
+        self.swap_chain = device.create_swap_chain(&self.surface, &self.sc_desc);
+    }
+    fn get_current_frame(&mut self) -> wgpu::SwapChainTexture {
+        self.swap_chain
+            .get_current_frame()
+            .expect("Failed to acquire next swap chain texture")
+            .output
+    }
+}
+
+async fn run(
+    event_loop: EventLoop<()>,
+    mut windows: Vec<Window>,
+    swapchain_format: wgpu::TextureFormat,
+) {
+    let instance = wgpu::Instance::new(wgpu::BackendBit::PRIMARY);
+    let first_window = windows.remove(0);
+    let first_surface = unsafe { instance.create_surface(&first_window) };
+    let adapter = instance
+        .request_adapter(&wgpu::RequestAdapterOptions {
+            power_preference: wgpu::PowerPreference::Default,
+            // Request an adapter which can render to our surface
+            compatible_surface: Some(&first_surface),
+        })
+        .await
+        .expect("Failed to find an appropiate adapter");
+
+    // Create the logical device and command queue
+    let (device, queue) = adapter
+        .request_device(
+            &wgpu::DeviceDescriptor {
+                features: wgpu::Features::empty(),
+                limits: wgpu::Limits::default(),
+                shader_validation: true,
+            },
+            None,
+        )
+        .await
+        .expect("Failed to create device");
+
+    let bind_group_layout = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+        label: None,
+        entries: &[
+            wgpu::BindGroupLayoutEntry {
+                binding: 0,
+                visibility: wgpu::ShaderStage::FRAGMENT,
```

Comment:
let's try to strip away all the parts of the example that are not related or critical to demonstrating the main aspect of it - multi-window. So there needs to be no shaders at all here: instead, you can just clear the color in each window to a different color (thus making a picture of windows?)

Reply:
All irrelevant parts are removed: texture creation, bind group and even pipeline.